### PR TITLE
parser, executor: support show/set session_states

### LINKED ISSUE #35259: support syntax `show/set session_states`
`SHOW SESSION_STATES` shows:
- the states on the current session, which is used to restore the states on the other session
- the token of the current session, which is used to connect the other TiDB instance with the same user

`SET SESSION_STATES ...` sets:
- the states showed from `SHOW SESSION_STATES`

These 2 statements are only used for the session manager.

### PR DESCRIPTION
**Pull request by djshow832**
### What problem does this PR solve?

Issue Number: close #35259

Problem Summary:
Supports empty `show session_states` and `set session_states`.
These syntaxes **WON'T** be exposed in documents.

### What is changed and how it works?

- the parser support `show session_states` and `set session_states`.
- the executor ignores the parameters and returns nothing.

### Check List

Tests 

- [x] Unit test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -6179,6 +6180,7 @@ TiDBKeyword:
 |	"PUMP"
 |	"SAMPLES"
 |	"SAMPLERATE"
+|   "SESSION_STATES"
```
Comment (lcwangchao) [inline parser/parser.y:6183]: This indent is a little strange

**Review by lcwangchao — COMMENTED** [review]
Rest LGTM

File: executor/executor_test.go [hunk 2]
```diff
@@ -215,6 +215,7 @@ func TestShow(t *testing.T) {
 		"RESTRICTED_REPLICA_WRITER_ADMIN Server Admin ",
 	))
 	require.Len(t, tk.MustQuery("show table status").Rows(), 1)
+	tk.MustQuery("show session_states").Check(testkit.Rows())
```
Comment (xhebox) [inline executor/executor_test.go:218]: Maybe just put it in `TestShowSessionStates`? Why distribute tests all around?

File: executor/simple_test.go [hunk 3]
```diff
@@ -1079,3 +1079,10 @@ func TestStatementsCauseImplicitCommit(t *testing.T) {
 		tk.MustExec("delete from ic")
 	}
 }
+
+func TestShowSessionStates(t *testing.T) {
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("set session_states 'x'")
```
Comment (xhebox) [inline executor/simple_test.go:1087]: Need at least one `MustNotExec` test.

File: executor/executor_test.go (hunk 2 above)
Comment (djshow832) [reply executor/executor_test.go:218]: `simple.go` covers `executeSetSessionStates` while `show.go` covers `fetchShowSessionStates`.
But I agree that putting them together makes code readable. I'll do it.

**Review by xhebox — APPROVED** [review]

**Review by lcwangchao — APPROVED** [review]

**Pull request merged**